service/controller/rbac/selector.go

```diff
@@ -0,0 +1,20 @@
+package rbac
+
+import (
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type wrongSelector struct {
+	labels.Selector
+	matchesFn func(labels.Labels) bool
+}
+
+func newWrongSelector(matchesFn func(labels.Labels) bool) wrongSelector {
+	return wrongSelector{
```

Comment:
I guess `Matches` is the only method that gets called on this object in `operatorkit`. If any of the others were to be called, then it would cause a nil pointer panic. I guess it's fine as long as we test all of the functions of `rbac-operator`.

Reply:
From a quick check, `Controller` only calls `Matches`. The selector is also passed into the controller-runtime client ListOptions for the timestamp metrics exporter which also seems to use `Matches` so it seems safe enough.

Reply:
Yes, that's why I named ith `wrongSelector`. I'll add some comments but before commenting it out I wanted to test if it works. And it looks like it does.